sidecar: port sidecar functionality to the 6.12 dev branch

benhillis opened the pull request:
This change pulls the sidecar functionality authored by @jstarks to the new 6.12 development branch. No functional changes from what is in 6.12, but had to resolve a few changes due to other mshv changes that have come in.

jstarks committed:
mshv_vtl: add sidecar kernel support

This adds support for the mshv_vtl sidecar kernel that runs on APs.

drivers/hv: don't fail to get the VTL

Some hypervisors (such as KVM) do not support the VsmVpStatus register.
Treat this the same as VTL so that kernels built with
CONFIG_HYPERV_VTL_MODE can still boot.

Signed-off-by: John Starks <jostarks@microsoft.com>

jstarks committed:
kernel: add boot_cpus command line parameter

Add a command line parameter, "boot_cpus", for specifying the set of
CPUs to online during boot.  CPUs that are not specified are still
marked present and can be onlined separately after boot.

This is essentially an extension of the existing maxcpus parameter.
maxcpus can limit the number of boot CPUs, but it always chooses the
first N CPUs (except on platforms with SMT enabled, where it will choose
the first thread of the first N cores).  This new parameter allows
specifying precisely which CPUs should start.  This is useful if the
online processors should include CPUs from specific NUMA nodes, for
device or memory locality reasons.

Without this change, a user can use maxcpus and then online the other
desired processors after boot. This is slow on large machines, and
it operates poorly with some drivers (e.g., those that make policy
decisions at startup based on cpu_online_mask). It is better to specify
the desired CPU set at boot.

Signed-off-by: John Starks <jostarks@microsoft.com>

jstarks commented on drivers/hv/hv_common.c:
```diff
@@ -340,8 +340,7 @@ u8 __init get_vtl(void)
 	} else {
-		pr_err("Failed to get VTL(error: %lld) exiting...\n", ret);
-		BUG();
+		ret = 0;
```
This change should be a separate commit.

Commit:
drivers/hv: don't fail to get the VTL

Some hypervisors (such as KVM) do not support the VsmVpStatus register.
Treat this the same as VTL so that kernels built with
CONFIG_HYPERV_VTL_MODE can still boot.

Signed-off-by: John Starks <jostarks@microsoft.com>
